packages/react/src/components/Tag/index.tsx

```diff
@@ -18,14 +20,37 @@ TagLabel.propTypes = {
   className: PropTypes.string
 };
 
-const Tag = ({ children, className, ...other }: TagProps) => (
-  <div className={classNames('Tag', className)} {...other}>
-    {children}
-  </div>
-);
+const Tag = ({
+  children,
+  className,
+  onToggle,
+  variant = 'default',
+  ...other
+}: TagProps) => {
+  if (variant === 'toggle') {
```

Comment:
I'm wondering if this should instead be `<Button variant="tag">` instead of `<Tag variant="toggle">`, that's similar to how we've done other variants is they're a variant of the semantic component that it's supposed to be.

There's a bit more nuance here because if we're treating this like a switch, we may want to consider having `aria-pressed` or `aria-checked` with `role="switch"` depending on the thing we're toggling. Because the allowed props are going to be dependent on the type of variant, and we need to allow props to be spread to the component. If we use `<Button variant="toggle">` we already have some of that built in for us.